packages/reactDom/test/test_RSC_html.ml

```diff
@@ -433,6 +433,31 @@ let await_tick ?(raise = false) num =
         let%lwt () = sleep ~ms:(Random.int 10) in
         if raise then Lwt.fail (Failure "lol") else Lwt.return (React.string num) )
 
+let server_function_as_action () =
+  let app () =
+    React.Upper_case_component
+      ( "app",
+        fun () ->
+          React.createElement "form"
+            [
+              React.JSX.Function
+                ( "action",
+                  "action",
+                  { Runtime.id = "1234-4321"; call = (fun () -> Lwt.return (React.string "Server Content")) } );
+            ]
+            [ React.string "Server Content" ] )
+  in
+  let main = React.Upper_case_component ("app", app) in
+  assert_html main ~disable_backtrace:true
+    ~shell:
+      "<form>Server Content</form><script data-payload='0:[\"$\",\"form\",null,{\"children\":[\"Server \
+       Content\"],\"action\":\"$F1\"},null,[],{}]\n\
```

Comment:
I remember discussing this in the past, and now being done is cool. But I don't remember if inlining the action id and action bound works here?

Reply:
It does not work, react uses the Model Reference ID (`$F`) to handle the function.

https://github.com/facebook/react/blob/a3be6829c6425f306a8bef9f7dba72d1347a64b3/packages/react-client/src/ReactFlightClient.js#L1485-L1495